pkg/virtualization/core/virtualsystem/virtualmachine.go

```diff
@@ -772,22 +772,37 @@ func (vm *VirtualMachine) GetDvdDrives() (col drive.DvdDriveCollection, err erro
 	return settings.GetVirtualDvdDrives()
 }
 
-func (vm *VirtualMachine) GetDvdDriveAndLogicalDiskByIsoPath(isoPath string) (dvd *drive.DvdDrive, diskdvd *disk.LogicalDisk, err error) {
+func (vm *VirtualMachine) GetDvdDriveAndLogicalDiskByIsoPath(isoPath string) (dvdDrive *drive.DvdDrive, logicalDisk *disk.LogicalDisk, err error) {
 	dvdCol, err := vm.GetDvdDrives()
 	if err != nil {
 		return
 	}
 	defer dvdCol.Close()
 
-	for _, inst := range dvdCol {
-		dvddisk, err1 := drive.GetRelatedStorageAllocationSettingData(inst.WmiInstance)
+	// make sure the return values are closed and nil in case of error
+	defer func() {
```

Comment:
@madhanrm we should be doing this for each iteration, right?

Reply:
This variable would be set only if the code crosses
if filepath.Clean(dvdPath) != filepath.Clean(isoPath) {

Once it crosses, it either finishes or errors out